README.md

```diff
@@ -1,94 +1,87 @@
-# The ops library
+Ops is a Python framework for developing and testing charms. It is an official component of the Charm SDK, itself a crucial part of the Juju universe.
+||||
+|-|-|- |
+|| Learn about Juju! | Start deploying charms: <br> https://juju.is/docs/juju <br> _Juju is declarative, model-driven, and has native support for integrations and scaling, on any cloud. You want to deploy `foo`? Run `juju deploy foo`! You wish to integrate `foo` and `bar`? Run `juju integrate foo bar`!_ |
+||||
+|| Get better at Juju! | Take advantage of the charm ecosystem: <br> https://charmhub.io/ <br> _Quickly deploy a cluster ([OpenStack](https://charmhub.io/openstack-base), [Kubernetes](https://charmhub.io/charmed-kubernetes)), a database ([PostgreSQL](https://charmhub.io/postgresql-k8s), [MongoDB](https://charmhub.io/mongodb), etc.), an observability stack ([Canonical Observability Stack](https://charmhub.io/cos-lite)), an MLOps solution ([Kubeflow](https://charmhub.io/kubeflow)), and so much more!_ |
+||||
+|:point_right:| **Make Juju your own!** | Write your own charm: <br> https://juju.is/docs/sdk <br> _Juju is written in Go, but our SDK supports easy charm development in Python!_ |
 
-<!-- The text below is also at the top of ops/__init__.py. Keep in sync! -->
+## Use Ops
+Give it a quick try:
 
-The ops library is a Python framework ([`available on PyPI`](https://pypi.org/project/ops/)) for developing
-and testing [Juju](https://juju.is/) charms in a consistent way, using standard Python constructs
-to allow for clean, maintainable, and reusable code.
+_(assumes you have a Linux that supports installation via snap)_
 
-A charm is an operator -- business logic encapsulated in a reusable software
-package that automates every aspect of an application's life.
-
-Charms written with ops support Kubernetes using Juju's "sidecar charm"
-pattern, as well as charms that deploy to Linux-based machines and containers.
-
-Charms should do one thing and do it well. Each charm drives a single
-application and can be integrated with other charms to deliver a complex
-system. A charm handles creating the application in addition to scaling,
-configuration, optimisation, networking, service mesh, observability, and other
-day-2 operations specific to the application.
-
-The ops library is part of the Charm SDK (the other part being Charmcraft).
-Full developer documentation for the Charm SDK is available at
-https://juju.is/docs/sdk.
-
-To learn more about Juju, visit https://juju.is/docs/olm.
-
-
-## Pure Python
+```
+# Use the Multipass 'charm-dev' blueprint to quickly get an Ubuntu VM with Charmcraft
+# as well as everything else you’ll need to develop and test a machine/ Kubernetes charm,
+# including `tox`, a LXD / Kubernetes cloud -- 'localhost' / 'microk8s' -- and Juju,
+# then use Juju to bootstrap a controller into a cloud -- below, 'microk8s'-
+# and add a model that you'll deploy your charm to.
```

Comment:
If there's going to be this mini tutorial in the readme then it would be better as regular marked-up text with bits of code than a big lump of code with comments.

I'm not convinced that having this form of tutorial makes sense here. It's not really an **ops** tutorial, it's more of a Juju one. Having a clear example of code built with the library in a readme is good, but this isn't really hitting that mark.

Reply:
Agreed on both points!

Reply:
Would it not be easier to just link to the dev setup docs on juju.is so there is only one place to update? https://juju.is/docs/sdk/dev-setup

Reply:
> If there's going to be this mini tutorial in the readme then it would be better as regular marked-up text with bits of code than a big lump of code with comments.

Happy to update that way.

> I'm not convinced that having this form of tutorial makes sense here. It's not really an ops tutorial, it's more of a Juju one. 

I don't think it is possible to give a full Ops tutorial here. Also, we do that in the docs. Here I mainly wanted to provide a quick way to experience Ops in context. As in, OK, you know about Juju, Charmhub, the SDK -- where does Ops fit in and what does it _look like_? PS The idea is to do the same for Juju and Charmcraft also. See https://github.com/juju/juju/pull/16478/files and https://github.com/canonical/charmcraft/pull/1341 (They're all very similar -- there's the same table at the top and a very similar "give it a quick try" tutorial featuring a Multipass charm-dev VM. However, the "You're here" pointer in the table and the focus in the tutorial change.)

> Having a clear example of code built with the library in a readme is good, but this isn't really hitting that mark.

I agree that it's very minimal. As I said, I don't believe it's realistic to give a full tutorial in the README, and we already do that in the SDK docs. My main hope here was to get people to the point where they'd at least take a look at some of the Ops code already there through Charmcraft init. Would it be better for you if, in addition to the install hook, we also highlighted something else,

> Would it not be easier to just link to the dev setup docs on juju.is so there is only one place to update? https://juju.is/docs/sdk/dev-setup

It would be. And I as a rule I would do everything I can to avoid duplication. However, https://juju.is/docs/sdk/dev-setup feels more like a how-to guide -- it focuses on the steps, gives options, it is not end to end. Here I was trying to go for a tutorial, which is why I wanted to make it focused on the results (as in, let's take the shortest path to where we can see some Ops), opinionated (as in, trust me and use the Multipass `charm-dev` VM), and end-to-end, even if with gaps and very minimal.

 (PS For more detail on why I decided for the Multipass `charm-dev` VM, see also https://github.com/canonical/charmcraft/pull/1341#issuecomment-1776892506 )

Reply:
Touched up the existing dev-setup doc to have a section specifically on spinning up a Multipass `charm-dev` machine: https://juju.is/docs/sdk/dev-setup#heading--automatic-set-up-an-ubuntu-charm-dev-vm-with-multipass I'll link to that instead